scipy/stats/_qmc.py

```diff
@@ -148,7 +186,12 @@ def scale(sample, l_bounds, u_bounds, *, reverse=False):
         return (sample - lower) / (upper - lower)
 
 
-def discrepancy(sample, *, iterative=False, method="CD", workers=1):
+def discrepancy(
+        sample: npt.ArrayLike,
+        *,
+        iterative: bool = False,
+        method: str = "CD",
+        workers: int = 1) -> float:
```

Comment:
Shouldn't we use `IntegerNumber` and `DecimalNumber` everywhere?

Reply:
Both `range` and `list` can currently take any object implementing the `__index__` protocol, 
so using `n: np.integer = ...` should be fine.

Note that the most recent mypy release still has the `range` input types defined `builtins.int`, a more
recent typeshed update has loosened their types to aforementioned protocol:
https://github.com/python/typeshed/blob/740f67245033a4e647706869a01f67d8c079fd7f/stdlib/builtins.pyi#L904-L911